src/model/Model.lua

```diff
@@ -168,7 +168,32 @@ end
      * @return the model text.
 ]]
 function Model:saveModelToText()
-
+	local tokenPatterns={}
+	for _,ptype in pairs({"r","p"}) do
+		for _,token in pairs(self.model[ptype][ptype].tokens) do
+			tokenPatterns[token]=string.gsub (string.gsub (token,"^p_","p."),"^r_","r.")
+		end
+	end
+	local s=""
+	local writeString=function(sec)
+		for ptype,_ in pairs(self.model[sec]) do
+			value=self.model[sec][ptype].value
+			for tokenPattern,newToken in pairs(tokenPatterns) do
+				value=string.gsub(value,tokenPattern,newToken)
+			end
+			s=s..sec.."="..value.."\n"
+		end
+	end
+	s=s.."[request_definition]\n"..writeString("r").."[policy_definition]\n"..writeString("p")
+	local _,ok=self.model["g"]
```

Comment:
I think this is Go syntax and it won't work here because `self.model["g"]` is a table and has only one value, please remove this.